cmd/storage-rest-server.go

```diff
@@ -897,10 +900,13 @@ func keepHTTPResponseAlive(w http.ResponseWriter) func(error) {
 			}
 		}
 		defer xioutil.SafeClose(doneCh)
+
 		ticker := time.NewTicker(time.Second * 10)
 		defer ticker.Stop()
 		for {
 			select {
+			case <-ctx.Done():
+				return
```

Comment:
There is a deadlock here. `doneCh` will block in the returned function. 

So below you will need something like:
```
		select {
			case doneCh <- err:
			case <-ctx.Done(): 
				doneCh = nil
				return
		}
```

`keepHTTPReqResponseAlive` above has a similar issue AFAICT.